### PR TITLE
Improves history management by using history.push

### PR DESCRIPTION
**Pull request by davidjoy**
We use history.push when adding a username or email, but generally use replace for errors and going back to /users

Also updating the document title with the search term.

### CODE REVIEW

File: src/users/UserPage.jsx [hunk 1]
```diff
@@ -62,7 +70,7 @@ export default function UserPage({ location }) {
         type: 'error',
         topic: 'general',
       });
-      history.replace('/users/');
+      history.replace('/users');
```
Comment (Dillon-Dumesnil) [inline src/users/UserPage.jsx:73]: Yeah, I noticed that when testing. Thanks for catching!

File: src/users/UserPage.jsx [hunk 2]
```diff
@@ -37,17 +37,25 @@ export default function UserPage({ location }) {
...
     }
+  }
 
+  function processSearchResult(searchValue, result) {
     if (result.errors.length > 0) {
       result.errors.forEach(error => add(error));
-      history.replace('/users/');
+      history.replace('/users');
+      document.title = 'Support Tools | edX';
+    } else if (isEmail(searchValue)) {
```
Comment (Dillon-Dumesnil) [inline src/users/UserPage.jsx:51]: Nice reorder

**Review by Dillon-Dumesnil — APPROVED** [review]
LGTM

**Review by staubina — APPROVED** [review]

File: src/users/UserPage.jsx [hunk 3]
```diff
@@ -37,17 +37,25 @@ export default function UserPage({ location }) {
     return !!(searchValue && searchValue.match(USERNAME_REGEX));
   }
 
-  function processSearchResult(searchValue, result) {
-    if (isEmail(searchValue)) {
-      history.replace(`/users/?email=${searchValue}`);
-    } else if (isValidUsername(searchValue)) {
-      history.replace(`/users/?username=${searchValue}`);
+  function pushHistoryIfChanged(nextUrl) {
+    if (nextUrl !== location.pathname + location.search) {
```
Comment (davidjoy) [inline src/users/UserPage.jsx:41]: I discovered that it creates a history entry even if the value is the same.

File: src/users/UserPage.jsx [hunk 4]
```diff
@@ -37,17 +37,25 @@ export default function UserPage({ location }) {
...
+      history.push(nextUrl);
     }
+  }
 
+  function processSearchResult(searchValue, result) {
     if (result.errors.length > 0) {
       result.errors.forEach(error => add(error));
-      history.replace('/users/');
+      history.replace('/users');
```
Comment (davidjoy) [inline src/users/UserPage.jsx:49]: In the case of errors, we just use replace.  That's not meaningful history.

File: src/users/UserPage.jsx [hunk 5]
```diff
@@ -114,13 +122,23 @@ export default function UserPage({ location }) {
     }
   }, [userIdentifier]);
 
+  useEffect(() => {
```
Comment (davidjoy) [inline src/users/UserPage.jsx:125]: This useEffect is necessary to get the page to re-search when the user uses the forward/back buttons.

**Pull request merged**